test/rbs/test/type_check_test.rb

```diff
@@ -69,6 +69,197 @@ def to_int: () -> Integer
+          assert typecheck.value(success, parse_type(interface))
+        rescue
+          p [:success, definer, interface]
+          raise
+        end
+
+        [
+          [ ->(o) { def o.f(a); end }, '::_NoArgs' ],
+          [ ->(o) { def o.f(a:); end }, '::_NoArgs' ],
+          [ ->(o) { def o.f(); end }, '::_PosArgs' ],
+          [ ->(o) { def o.f(a, b); end }, '::_PosArgs' ],
+          [ ->(o) { def o.f(a, b, *r); end }, '::_PosArgs' ],
+          [ ->(o) { def o.f(a: 1); end }, '::_PosArgs' ],
+          [ ->(o) { def o.f(a, &b); end }, '::_OptArgs' ],
+          [ ->(o) { def o.f(); end }, '::_PosOptArgs' ],
+          [ ->(o) { def o.f(a); end }, '::_PosOptArgs' ],
+          [ ->(o) { def o.f(a, b); end }, '::_PosOptArgs' ],
+          [ ->(o) { def o.f(); end }, '::_RestArgs' ],
+          [ ->(o) { def o.f(a, *r); end }, '::_RestArgs' ],
+          [ ->(o) { def o.f(*r, a); end }, '::_RestArgs' ],
+          [ ->(o) { def o.f(a); end }, '::_TrailingArgs' ],
+          [ ->(o) { def o.f(a, *b, c); end }, '::_TrailingArgs' ],
+          [ ->(o) { def o.f(); end }, '::_ReqKeyArgs' ],
+          [ ->(o) { def o.f(a:, b:, c:); end }, '::_ReqKeyArgs' ],
+          [ ->(o) { def o.f(a:); end }, '::_ReqKeyArgs' ],
+          [ ->(o) { def o.f(a: 1); end }, '::_ReqKeyArgs' ],
+          [ ->(o) { def o.f(); end }, '::_OptKeyArgs' ],
+          [ ->(o) { def o.f(a); end }, '::_OptKeyArgs' ],
+          [ ->(o) { def o.f(a:, b: 'b'); end }, '::_OptKeyArgs' ],
+          [ ->(o) { def o.f(a); end }, '::_PosReqKeyArgs' ],
+          [ ->(o) { def o.f(a, z:); end }, '::_PosReqKeyArgs' ],
+          [ ->(o) { def o.f(); end }, '::_RestKeyArgs' ],
+          [ ->(o) { def o.f(a); end }, '::_RestKeyArgs' ],
+          [ ->(o) { def o.f(a:); end }, '::_RestKeyArgs' ],
+          [ ->(o) { def o.f(a:, **rk); end }, '::_RestKeyArgs' ],
+          [ ->(o) { def o.f(); end }, '::_ReqKeyRestKeyArgs' ],
+          [ ->(o) { def o.f(a:); end }, '::_ReqKeyRestKeyArgs' ],
+          [ ->(o) { def o.f(a, **b); end }, '::_ReqKeyRestKeyArgs' ],
+          [ ->(o) { def o.f(**rk); end }, '::_PosRestKeyArgs' ],
+          [ ->(o) { def o.f(); end }, '::_AllArgs' ],
+          [ ->(o) { def o.f(**rk); end }, '::_AllArgs' ],
+          [ ->(o) { def o.f(a, b); end }, '::_Overload' ],
+          [ ->(o) { def o.f(a); end }, '::_Overload' ],
+          [ ->(o) { def o.f(a = nil); end }, '::_Overload' ],
+          [ ->(o) { def o.f(); end }, '::_Overload' ],
+          [ ->(o) { def o.f(a, b, c); end }, '::_Overload' ],
+        ].each do |definer, interface|
+          failure = Object.new
+          definer.call(failure)
+
+          no = ArgumentChecker.new(builder: builder, interface: interface).no_argument_error?(:f) do |args, kwargs, block|
+            failure.f(*args, **kwargs, &block)
+          end
+          refute no
+          refute typecheck.value(failure, parse_type(interface))
+        rescue
+          p [:failure, definer, interface]
```

Comment:
Is this intentionally print using `p`? (Not sure if it should be.)

Reply:
When a test failed, it was intentionally included to indicate which line of the case had failed.
However, I agree that it is complicated and will remove it.